java/ql/src/experimental/Security/CWE/CWE-117/examples/LogInjection.java

```diff
@@ -0,0 +1,33 @@
+package com.example.restservice;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class LogInjection {
+
+    private final Logger log = LoggerFactory.getLogger(LogInjection.class);
+
+    // /good?username=Guest'%0AUser:'Admin
+    @GetMapping("/good")
+    public String good(@RequestParam(value = "username", defaultValue = "name") String username) {
+        username = username.replace("\n", "");
+        log.warn("User:'{}'", username);
+        return new String(username);
```

Comment:
Why return a new string (same in line 27)? `String` is immutable so there is no need to create a copy

Reply:
You are right. I was doing some experiments and I forgot to change the return type. I made the change. Thanks for all the other feedback. I really appreciated. I will do the other changes/suggestions as soon as I will have some time (unfortunately I am a little bit busy at the moment).